analysis/deep2_r23_o32/zoom_and_gauss_general.py

```diff
@@ -367,98 +392,109 @@ def zoom_gauss_plot(dataset, tab, stack2d, dispersion, s2, wave,
-        if line_type[ii] == 'Balmer': 
-            outpdf = fitspath + dataset + '_Zoomed_Gauss_' + line_name[ii] + '.pdf'
-            print(outpdf)
-            em_tab, R_23_array, O_32_array, N_gal_array, id = zoom_gauss_plot(dataset,
-                                                                              tab, stack2d, dispersion, s2,
-                                                                              wave, lambda0[ii], lineflag,
-                                                                              y_correction=y_correction,
-                                                                              line_type=line_type[ii],
-                                                                              outpdf=outpdf, line_name=line_name[ii])
+        if line_type[ii] == 'Balmer':
+            em_tab, R_23_array, O_32_array, N_gal_array, \
+                id = zoom_gauss_plot(dataset, tab, stack2d, dispersion, s2,
+                                     wave, lambda0[ii], lineflag,
+                                     y_correction=y_correction,
+                                     line_type=line_type[ii],
+                                     outpdf=outpdf, line_name=line_name[ii],
+                                     log=log)
 
         # Oxy2 Line Fit
-        if line_type[ii] == 'Oxy2': 
-            outpdf = fitspath + dataset + '_Zoomed_Gauss_' + line_name[ii] + '.pdf'
-            print(outpdf)
-            em_tab, R_23_array, O_32_array, N_gal_array, id = zoom_gauss_plot(dataset,
-                                                                              tab, stack2d, dispersion, s2,
-                                                                              wave, lambda0[ii], lineflag,
-                                                                              y_correction=y_correction,
-                                                                              line_type=line_type[ii],
-                                                                              outpdf=outpdf, line_name=line_name[ii])
+        if line_type[ii] == 'Oxy2':
+            em_tab, R_23_array, O_32_array, N_gal_array, \
+                id = zoom_gauss_plot(dataset, tab, stack2d, dispersion, s2,
+                                     wave, lambda0[ii], lineflag,
+                                     y_correction=y_correction,
+                                     line_type=line_type[ii],
+                                     outpdf=outpdf, line_name=line_name[ii],
+                                     log=None)
 
         if ii == 0:
-            out_ascii_single = fitspath + '/' + dataset + '_Average_R23_O32_Values.tbl'
-
             n2 = ('bin_ID', 'logR23_avg', 'logO32_avg', 'N_stack')
-            avg_tab = Table([id, R_23_array, O_32_array, N_gal_array], names=n2)
-            
+            avg_tab = Table([id, R_23_array, O_32_array, N_gal_array],
+                            names=n2)
+
```

Comment:
@Reagen: I'm not sure why we need to output `R23_array`, etc from `zoom_gauss_plot`. These are the grid values right? They should be retrieved separately and pass in before looping over each line. The only thing from `zoom_gauss_plot` should be `em_tab`.

Reply:
Ok, I saw you opened up a separate issue for this?

Reply:
Correct #65